modules/react/src/components/export-video/export-video-panel-container.js

```diff
@@ -44,64 +44,82 @@ import {getQualitySettings, HIGH_16_9} from './constants';
 //     .toString()}`;
 // }
 
-export class ExportVideoPanelContainer extends Component {
-  static defaultProps = {
-    exportVideoWidth: 980
-  };
+const ENCODERS = {
+  gif: GifEncoder,
+  webm: WebMEncoder,
+  jpeg: JPEGSequenceEncoder,
+  png: PNGSequenceEncoder
+};
 
+export class ExportVideoPanelContainer extends Component {
   constructor(props) {
     super(props);
 
-    this.setMediaTypeState = this.setMediaTypeState.bind(this);
+    this.setMediaType = this.setMediaType.bind(this);
     this.setCameraPreset = this.setCameraPreset.bind(this);
     this.setFileName = this.setFileName.bind(this);
-    this.setQuality = this.setQuality.bind(this);
+    this.setResolution = this.setResolution.bind(this);
     this.getCameraKeyframes = this.getCameraKeyframes.bind(this);
     this.getDeckScene = this.getDeckScene.bind(this);
     this.onPreviewVideo = this.onPreviewVideo.bind(this);
     this.onRenderVideo = this.onRenderVideo.bind(this);
     this.setDuration = this.setDuration.bind(this);
 
     this.state = {
-      mediaType: 'GIF',
+      mediaType: 'gif',
       cameraPreset: 'None',
       fileName: 'Video Name',
-      qualitySettings: HIGH_16_9,
+      resolution: 1,
+      durationMs: 1000, // TODO change to 5000 later. 1000 for dev testing
       viewState: this.props.mapData.mapState,
-      durationMs: 1000,
-      canvasWidth: 1280, // canvas size changes resolution for everything but GIF (? unsure)
-      canvasHeight: 720,
-      encoderSettings: {
-        framerate: 30,
-        webm: {
-          quality: 0.8
-        },
-        jpeg: {
-          quality: 0.8
-        },
-        gif: {
-          sampleInterval: 1000,
-          width: 1280,
-          height: 720
-        },
-        filename: 'kepler.gl'
-      },
+      filename: 'kepler.gl',
       adapter: new DeckAdapter(this.getDeckScene)
     };
   }
 
+  getCanvasSize() {
+    const {resolution} = this.state;
+    return getResolutionSetting(resolution);
+  }
+
+  getEncoderSettings() {
```

Comment:
Encoder settings were originally in state since we were going to eventually expose advanced settings tweaks. But that is out of scope for now and this works great. 

Besides, directly tweaking these settings may be more then a user needs - we could instead consider a few "quality" settings that translate to each encoder's unique needs. GIF in particular has many more options to tweak (a second-pass of scaling and framerate, as well a few compression options)

Reply:
Well even if you have some per-encoder settings, this pattern of composing the derived state explicitly still works better in the long run imo.

e.g. You could still have `state.gif` a merge a few fields in as `...state.gif`.